osxkeychain.go

```diff
@@ -46,41 +49,39 @@ type InternetPassword struct {
 	AuthType       AuthenticationType
 }
 
-// Error codes from https://developer.apple.com/library/mac/documentation/security/Reference/keychainservices/Reference/reference.html#//apple_ref/doc/uid/TP30000898-CH5g-CJBEABHG
-var (
-	ErrUnimplemented     = errors.New("Function or operation not implemented.")
-	ErrParam             = errors.New("One or more parameters passed to the function were not valid.")
-	ErrAllocate          = errors.New("Failed to allocate memory.")
-	ErrNotAvailable      = errors.New("No trust results are available.")
-	ErrReadOnly          = errors.New("Read only error.")
-	ErrAuthFailed        = errors.New("Authorization/Authentication failed.")
-	ErrNoSuchKeychain    = errors.New("The keychain does not exist.")
-	ErrInvalidKeychain   = errors.New("The keychain is not valid.")
-	ErrDuplicateKeychain = errors.New("A keychain with the same name already exists.")
-	ErrDuplicateCallback = errors.New("More than one callback of the same name exists.")
-	ErrInvalidCallback   = errors.New("The callback is not valid.")
-	ErrDuplicateItem     = errors.New("The item already exists.")
-	ErrItemNotFound      = errors.New("The item cannot be found.")
+type _OSStatus C.OSStatus
+
+// TODO: Fill this out.
+const (
+	errDuplicateItem _OSStatus = C.errSecDuplicateItem
```

Comment:
Yes! So much better. I had forgotten that `error` was an interface and we just had to make a type with an `Error() string` method to satisfy it.

I was thinking that this type should be exported in some way so that the user could get the error code back, but thinking about it more I'm not sure that's necessary. We'll deal with it if there's a compelling reason to expose it.